addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -184,6 +187,16 @@ def write(self, vals):
     def unlink(self):
         # sudo: discuss.channel.rtc.session - cascade unlink of sessions for self member
         self.sudo().rtc_session_ids.unlink()  # ensure unlink overrides are applied
+        # if the only owner of the current channel is removed
+        for member in self.filtered(
+            lambda m: m == m.channel_id.channel_member_ids.filtered("is_owner")
+        ):
+            # set the first member with user_ids as owner
+            member_with_user_ids = member.channel_id.channel_member_ids.filtered(
+                lambda m: m.partner_id.user_ids and not m.is_owner
+            )
+            if member_with_user_ids:
+                member_with_user_ids.sorted("id")[0].is_owner = True
```

Comment:
All of those should be made with specific queries, using x2many fields does not scale with a lot of members as it needs to load them all.